paper_framework_sources/p1897_initial_executor_algorithms_for_cpp23.md

```diff
@@ -587,42 +641,47 @@ The expression `execution::indexed_for(S, P, R, F)` for some subexpressions `S`,
 
 Signature:
 ```cpp
-S<T2> transform(S<T...>, invocable<T2(T...));
+template <execution::sender S, std::invocable F>
+see-below transform(S s, F f);
 ```
 
-where `S<T...>` and `S<T2>` are implementation-defined types that is represent senders that send a value of type list `T...` or `T2` respectively in their value channels.
-Note that in the general case there may be many types `T...` for a given `sender`, in which case the invocable may have to represent an overload set.
-
 *[ Example:*
 ```cpp
 int r = sync_wait(just(3) | transform([](int v){return v+1;}));
 // r==4
 ```
+*- end example]*
 
 ### Wording
+
 The name `execution::transform` denotes a customization point object.
-The expression `execution::transform(S, F)` for some subexpressions `S` and `F` is expression-equivalent to:
+For some subexpressions `s` and `f`, let `S` be a type such that `decltype((s))` is `S` and `decltype((f))` is `F`.
+The expression `execution::transform(s, f)` is expression-equivalent to:
 
- * `S.transform(F)` if that expression is valid.
- * Otherwise, `transform(S, F)`, if that expression is valid with overload resolution performed in a context that includes the declaration
+ * `s.transform(f)` if that expression is valid, `s` satisfies `sender` and `f` satisfies `invocable`.
```

Comment:
`invocable` is the wrong constraint here. That requires that `f` is callable with no arguments. You can really only check the type of `f` here if `s` is a `typed_sender`. The constraint on `f` is typically checked on the `set_value` of `transform`'s receiver.